src/pybind/mgr/dashboard/tests/test_tools.py

```diff
@@ -84,11 +89,11 @@ def test_fill(self):
 
     def test_not_implemented(self):
         self._put("/foo")
-        self.assertStatus(405)
+        self.assertStatus(404)
```

Comment:
`Method Not Implemented` would be the correct status code here. This would also break the browsable API for controllers that don't implement `list` or `__call__`

Reply:
Since the new dispatcher has now all possible URL patterns plus their allowed HTTP methods that are available by our code, the dispatcher returns 404 for all URLs that do not match the requirements.

If you think it's worth it, I can disable the filtering of "allowed methods" by the dispatcher and do that check in the `RESTController` class where we were doing previously, that way we can return 405 in these cases.